test/inspector/inspector-helper-new.js

```diff
@@ -0,0 +1,427 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+const http = require('http');
+const path = require('path');
+const { spawn } = require('child_process');
+const url = require('url');
+
+const MAIN_SCRIPT = path.join(common.fixturesDir, 'loop.js');
+const DEBUG = false;
+const TIMEOUT = 15 * 1000;
+
+function makeBufferingDataCallback(dataCallback) {
+  let buffer = Buffer.alloc(0);
+  return (data) => {
+    const newData = Buffer.concat([buffer, data]);
+    const str = newData.toString('utf8');
+    const lines = str.split('\n');
+    if (str.endsWith('\n'))
+      buffer = Buffer.alloc(0);
+    else
+      buffer = Buffer.from(lines.pop(), 'utf8');
+    for (const line of lines)
+      dataCallback(line);
+  };
+}
+
+function tearDown(child, err) {
+  child.kill();
+  if (err) {
+    console.error(err);
+    process.exit(1);
+  }
+}
+
+function checkHttpResponse(host, port, path) {
+  return new Promise(function(resolve, reject) {
+    const req = http.get({ host, port, path }, function(res) {
+      let response = '';
+      res.setEncoding('utf8');
+      res
+        .on('data', (data) => response += data.toString())
+        .on('end', () => {
+          resolve(response);
+        });
+    });
+    req.on('error', reject);
+  });
+}
+
+function timeout(message, multiplicator) {
+  return setTimeout(common.mustNotCall(message),
+                    TIMEOUT * (multiplicator || 1));
+}
+
+function parseWSFrame(buffer) {
+  let message = null;
+  if (buffer.length < 2)
+    return { length: 0, message };
+  if (buffer[0] === 0x88 && buffer[1] === 0x00) {
+    return { length: 2, message };
+  }
+  assert.strictEqual(0x81, buffer[0]);
+  let dataLen = 0x7F & buffer[1];
+  let bodyOffset = 2;
+  if (buffer.length < bodyOffset + dataLen)
+    return 0;
+  if (dataLen === 126) {
+    dataLen = buffer.readUInt16BE(2);
+    bodyOffset = 4;
+  } else if (dataLen === 127) {
+    dataLen = buffer.readUInt32BE(2);
+    bodyOffset = 10;
+  }
+  if (buffer.length < bodyOffset + dataLen)
+    return { length: 0, message };
+  message = JSON.parse(
+    buffer.slice(bodyOffset, bodyOffset + dataLen).toString('utf8'));
+  if (DEBUG)
+    console.log('[received]', JSON.stringify(message));
+  return { length: bodyOffset + dataLen, message };
+}
+
+function promiseWithTimeout(promise, message, multiplicator) {
```

Comment:
I copied you version - but the issue I noticed is that the test will not complete until the timeout is fired. I implemented the timeout clearing.